sklearn/metrics/classification.py

```diff
@@ -211,6 +212,10 @@ def confusion_matrix(y_true, y_pred, labels=None, sample_weight=None):
         If none is given, those that appear at least once
         in ``y_true`` or ``y_pred`` are used in sorted order.
 
+    normalize : bool, optional (default=False)
+        If True, , return the fraction of classified samples (float),
```

Comment:
You can normalise by number of samples (like accuracy), by number of samples in the ground truth for each class (like recall), or by number of samples predicted for each class (like precision). I find this description highly ambiguous, and I'm not persuaded we should assume one normalisation is more appropriate than another for the user.

Reply:
Good point. I did it that way because most frequently the normalization is done with regard to the ground truth. When we think of a normalized confusion matrix the picture that comes to mind precisely something like [this](https://scikit-learn.org/stable/_images/sphx_glr_plot_confusion_matrix_002.png).

To avoid suggesting to the user just one specific type of normalization, different arguments for the `normalize` parameter could be specified, e.g, "precision/accuracy/recall".